### PR TITLE
Use 'markdownDescription' for property completion

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Use 'markdownDescription' for property completion.
Demo:
Screenshot 2021-03-25 at 12 05 03

### What issues does this PR fix or reference?
Fix: https://github.com/redhat-developer/vscode-yaml/issues/417

### Is it tested? How?
With test

### CODE REVIEW

File: test/autoCompletion.test.ts [hunk 1]
```diff
@@ -1840,6 +1840,33 @@ describe('Auto Completion Tests', () => {
...
+            type: 'string',
+            description: 'Kind is a string value representing the REST',
+            markdownDescription:
+              '**kind** (string)\n\nKind is a string value representing the REST resource this object represents.',
+          },
+        },
+        required: ['kind'],
+      });
+
+      const content = 'kin';
```
Comment (joshuawilson) [inline test/autoCompletion.test.ts:1858]: Should `kin` be `kind`?

Comment (evidolob) [reply test/autoCompletion.test.ts:1858]: It should be `kin` to test completion to `kind`

Comment (joshuawilson) [reply test/autoCompletion.test.ts:1858]: ok

**Review by joshuawilson — APPROVED** [review]

**Review by JPinkney — COMMENTED** [review]
I think we might have to add a `CompletionClientCapabilities` check for the documentationFormat. If I'm understanding the spec correctly, under https://microsoft.github.io/language-server-protocol/specification#textDocument_completion it looks like clients can specify what documentation formats they support in `completionItem.documentationFormat`

**Comment by evidolob** [comment]
@JPinkney In all cases we use [fromMarkup](https://github.com/microsoft/vscode-json-languageservice/blob/main/src/services/jsonCompletion.ts#L958) Which already has such check.
I will double check that we use that method in all cases.

**Pull request merged**